app/scripts/controllers/preferences.js

```diff
@@ -480,11 +506,8 @@ export default class PreferencesController {
         let addressBookKey = rpcDetail.chainId;
         if (!addressBookKey) {
           // We need to find the networkId to determine what these addresses were keyed by
-          const provider = new ethers.providers.JsonRpcProvider(
```

Comment:
This change is okay because networkcontroller.setrpc is called before `updateRpc`